src/material/table/testing/table-harness.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {TableHarnessFilters, RowHarnessFilters} from './table-harness-filters';
+import {MatRowHarness, MatHeaderRowHarness, MatFooterRowHarness} from './row-harness';
+
+/** Data extracted from a table organized by columns. */
+export interface MatTableHarnessColumnsText {
+  [columnName: string]: {
+    text: string[];
+    headerText: string;
```

Comment:
I assume its possible to not have a header or footer row? so maybe `?: string` would be better. Or if it's also possible to have multiple header/footer rows then `string[]`

Reply:
It is possible to have more than one header/footer row, but I decided to pick out the first one and consider it the primary one. Having an array of headers/footers feels like it would be cumbersome to deal with.